controllers/v1/observationSubmissionsController.js

```diff
@@ -113,29 +115,39 @@ module.exports = class ObservationSubmissions extends Abstract {
 
         observationDocument = observationDocument[0];
 
-        let entityDocument = await entitiesHelper.entityDocuments({
-          _id: req.query.entityId,
-          entityType: observationDocument.entityType
-        }, [
-          "metaInformation",
-          "entityTypeId",
-          "entityType",
-          "registryDetails"
-        ]);
+        let entityResult = [];
+        let filterData = {
+            "id" : req.query.entityId,
+            "type" : observationDocument.entityType
+        };
 
-        if (!entityDocument[0]) {
-          return resolve({ 
-            status: httpStatusCode.bad_request.status, 
-            message: messageConstants.apiResponses.ENTITY_NOT_FOUND
-          });
+        let entitiesDocument = await sunbirdService.learnerLocationSearch( filterData );
+        
+        if ( !entitiesDocument.success || !entitiesDocument.data.response.length > 0 ) {
+            let responseMessage = messageConstants.apiResponses.ENTITY_NOT_FOUND;
+            return resolve({ 
+                status: httpStatusCode.bad_request.status, 
+                message: responseMessage 
+            });
         }
+        entityResult.push(entitiesDocument.data.response);
         
-        entityDocument = entityDocument[0];
+        let entityDocument = {
+            id : entityResult[0].id,
+            registryDetails : {
+                locationId : entityResult[0].id,
+                code : entityResult[0].code
+            },
+            entityType : entityResult[0].type,
+            metaInformation : {
+                name : entityResult[0].name,
+                entityExternalId : entityResult[0].code
+            }
+        };
 
         if (entityDocument.registryDetails && Object.keys(entityDocument.registryDetails).length > 0) {
```

Comment:
@VISHNUDAS-tunerlabs remove this if condition check registryDetails will be there always if you are getting data from sunbird search